src/init/init.ts

```diff
@@ -0,0 +1,181 @@
+}
+
+function getGeneratorDescription(generator: YeomanEnvironment.GeneratorMeta, generatorName: string): GeneratorDescription {
+  let name = getDisplayName(generatorName);
+  let description;
+
+  if (templateDescriptions.hasOwnProperty(name)) {
+    description = templateDescriptions[name];
+  } else if (generator.resolved && generator.resolved !== 'unknown') {
+    try {
+      let metapath = findup.sync(generator.resolved, 'package.json');
+      let meta = JSON.parse(fs.readFileSync(metapath, 'utf8'));
+      description = meta.description;
+    } catch (err) {
+      if (err.message === 'not found') {
+        logger.debug('no package.json found for generator');
+      } else {
+        logger.debug('problem reading/parsing package.json for generator', {
+          generator: generatorName,
+          err: err.message,
+        });
+      }
+      description = '';
+    }
+  }
+
+  return {
+    name: `${name}  ${chalk.dim(description)}`,
+    value: generatorName,
+    // inquirer is broken and doesn't print descriptions :(
+    // keeping this so things work when it does
+    short: name,
+  };
+}
+
+function getDisplayName(generatorName: string) {
+  let nameEnd = generatorName.indexOf(':');
+  if (nameEnd === -1) {
+    nameEnd = generatorName.length;
+  }
+  return generatorName.substring('polymer-init-'.length, nameEnd);
+}
+
+export function createYeomanEnvironment(): Promise<any> {
+  return new Promise((resolve, reject) => {
+    let env = new YeomanEnvironment();
+    env.registerStub(ElementGenerator, 'polymer-init-element:app');
+    env.registerStub(ApplicationGenerator, 'polymer-init-application:app');
+    env.registerStub(shopGenerator, 'polymer-init-shop:app');
+    env.registerStub(appDrawerGenerator, 'polymer-init-app-drawer-template:app');
+    env.lookup((err) => {
+      if (err) {
+        reject(err);
+        return;
+      }
+      resolve(env);
+    });
+  });
+}
+
+export function runGenerator(generatorName, options): Promise<any> {
+  options = options || {};
+  let templateName = options.templateName || generatorName;
+
+  return new Promise((resolve, reject) => {
+    resolve(options.env || createYeomanEnvironment());
+  }).then(function(env) {
+    let generators = env.getGeneratorsMeta();
+    let generator = generators[generatorName];
+    if (!generator) {
+      logger.error(`Template ${templateName} not found`);
+      throw new Error(`Template ${templateName} not found`);
+    }
+
+    logger.info(`Running template ${options.templateName || generatorName}`);
+    logger.debug(`Running generator ${generatorName}`);
+    return new Promise((resolve, reject) => {
+      env.run(generatorName, {}, () => resolve());
+    });
+  });
+}
+
+export function promptGeneratorSelection(options): Promise<any> {
+  options = options || {};
+
+  return new Promise((resolve, reject) => {
+    resolve(options.env || createYeomanEnvironment());
+  }).then(function(env) {
+    let generators = env.getGeneratorsMeta();
```

Comment:
Factor this out into a simple function that generates the template list. It'll be easier to test that.